flink-connectors/flink-hbase/src/test/java/org/apache/flink/addons/hbase/example/HBaseTableSourceITCase.java

```diff
@@ -0,0 +1,196 @@
+
+		String expected = "100,strvalue,19991\n" +
+			"101,strvalue1,19992\n" +
+			"102,strvalue2,19993\n";
+		TestBaseUtils.compareResultAsText(results, expected);
+	}
+
+	@Test
+	public void testHBaseTableSourceWithTwoColumnFamily() throws Exception {
+		// create a table with single region
+		TableName tableName = TableName.valueOf("test1");
+		// no split keys
+		byte[][] famNames = new byte[2][];
+		famNames[0] = F_1;
+		famNames[1] = F_2;
+		createTable(tableName, famNames, null);
+		// get the htable instance
+		HTable table = openTable(tableName);
+		List<Put> puts = new ArrayList<Put>();
+		// add some data
+		Put put = new Put(ROW_1);
+		// add 3 qualifiers per row
+		//1st qual is integer
+		put.addColumn(F_1, Q_1, Bytes.toBytes(100));
+		//2nd qual is String
+		put.addColumn(F_1, Q_2, Bytes.toBytes("strvalue"));
+		// 3rd qual is long
+		put.addColumn(F_1, Q_3, Bytes.toBytes(19991L));
+		puts.add(put);
+
+		put = new Put(ROW_2);
+		// add 3 qualifiers per row
+		//1st qual is integer
+		put.addColumn(F_2, Q_1, Bytes.toBytes(201));
+		//2nd qual is String
+		put.addColumn(F_2, Q_2, Bytes.toBytes("newvalue1"));
+		// 3rd qual is long
+		put.addColumn(F_2, Q_3, Bytes.toBytes(29992L));
+		puts.add(put);
+
+		put = new Put(ROW_3);
+		// add 3 qualifiers per row
+		//1st qual is integer
+		put.addColumn(F_1, Q_1, Bytes.toBytes(102));
+		//2nd qual is String
+		put.addColumn(F_1, Q_2, Bytes.toBytes("strvalue2"));
+		// 3rd qual is long
+		put.addColumn(F_1, Q_3, Bytes.toBytes(19993L));
+		puts.add(put);
+		// add the mutations to the table
+		table.put(puts);
+		table.close();
+		ExecutionEnvironment env = ExecutionEnvironment.getExecutionEnvironment();
+		BatchTableEnvironment tableEnv = TableEnvironment.getTableEnvironment(env, 	new TableConfig());
+		HBaseTableSchema schema = new HBaseTableSchema();
+		schema.addColumn(Bytes.toString(F_1), Bytes.toString(Q_1), Integer.class);
+		schema.addColumn(Bytes.toString(F_1), Bytes.toString(Q_2), String.class);
+		schema.addColumn(Bytes.toString(F_1), Bytes.toString(Q_3), Long.class);
+		schema.addColumn(Bytes.toString(F_2), Bytes.toString(Q_1), Integer.class);
+		schema.addColumn(Bytes.toString(F_2), Bytes.toString(Q_2), String.class);
+		schema.addColumn(Bytes.toString(F_2), Bytes.toString(Q_3), Long.class);
+		// fetch row2 from the table till the end
+		BatchTableSource hbaseTable = new HBaseTableSource(getConf(), tableName.getNameAsString(), schema);
+		tableEnv.registerTableSource("test1", hbaseTable);
+		Table result = tableEnv
+			.sql("SELECT test1.f1.q1, test1.f1.q2, test1.f1.q3, test1.f2.q1, test1.f2.q2, test1.f2.q3 FROM test1");
+		DataSet<Row> resultSet = tableEnv.toDataSet(result, Row.class);
+		List<Row> results = resultSet.collect();
+
+		String expected = "100,strvalue,19991,null,null,null\n" +
+			"null,null,null,201,newvalue1,29992\n" +
+			"102,strvalue2,19993,null,null,null\n";
+		TestBaseUtils.compareResult(results, expected, false, false);
```

Comment:
Actually this API was expecting things to be sorted. I did not want that to happen. Ok then will ensure that the assertion is done after sorting. Thanks for all the comments. Will update the PR shortly.